### PR TITLE
Fix part #23: Hi fi topic multiple tabs

### PR DESCRIPTION
**Pull request by nikitamarysolomanpvt**
The topic page should provide background information on the topic, skills corresponding to the topic (which link to the concept card for that skill), navigation to training on skills in the topic, the progress of stories the user has played in that topic, and links to stories (with preference for in-progress/immediately upcoming stories, though all stories should be viewable). See the PRD for specifics.

Note that this is tracking implementing the final UI for just the topic page. The final UIs for each constituent tab is indicated as a blocker of this issue.

https://xd.adobe.com/spec/e2239cf4-9cde-4c08-5296-25316c1f0a14-9412/screen/19cfbacf-854c-4c7d-8691-3b3d117e1866/TP-Overview-

## Task Items
- [x]  Toolbar/ActionBar title
Screenshot 2019-11-05 at 9 26 38 PM

- [x] TabLayout with icons and titles
Screenshot 2019-11-05 at 9 26 45 PM

- [x] Final check on all fragments and final touch-up

- [x] Accessibility Scanner is working correctly (https://support.google.com/accessibility/android/answer/6376570?hl=en)

## Checklist

- [x] The PR title starts with "Fix #bugnum: ", followed by a short, clear summary of the changes. (If this PR fixes part of an issue, prefix the title with "Fix part of #bugnum: ...".)
- [x] The PR explanation includes the words "Fixes #bugnum: ..." (or "Fixes part of #bugnum" if the PR only partially fixes an issue).
- [x] The PR follows the [style guide](https://github.com/oppia/oppia-android/wiki/Coding-style-guide).
- [x] The PR does not contain any unnecessary auto-generated code from Android Studio.
- [x] The PR is made from a branch that's **not** called "develop".
- [x] The PR is made from a branch that is up-to-date with "develop".
- [x] The PR's branch is based on "develop" and not on any other branch.
- [x] The PR is **assigned** to an appropriate reviewer in both the **Assignees** and the **Reviewers** sections.

### CODE REVIEW

File: app/src/main/res/values/styles.xml [hunk 1]
```diff
@@ -79,4 +79,22 @@
...
+    <item name="tabIndicatorColor">@color/white</item>
+    <item name="tabIndicatorHeight">4dp</item>
+    <item name="tabPaddingStart">6dp</item>
+    <item name="tabPaddingEnd">6dp</item>
+    <item name="tabBackground">?attr/selectableItemBackground</item>
+    <item name="tabTextAppearance">@style/AppTabTextAppearance</item>
+    <item name="tabIconTint">@color/tab_icon_color_selector</item>
+    <item name="tabSelectedTextColor">@color/white</item>
+  </style>
+  <!-- for text -->
```
Comment (rt4914) [inline app/src/main/res/values/styles.xml]: I think we don't need this comment, the style name is self explanatory.

File: app/src/main/res/values/styles.xml [hunk 2]
```diff
@@ -79,4 +79,22 @@
...
+  <style name="AppTabTextAppearance" parent="TextAppearance.Design.Tab">
+    <item name="android:textSize">14sp</item>
+    <item name="android:textColor">@color/tab_icon_color_selector</item>
+    <item name="android:fontFamily">sans-serif</item>
```
Comment (rt4914) [inline app/src/main/res/values/styles.xml]: font-family is `sans-serif-medium`

File: app/src/main/res/layout/topic_fragment.xml [hunk 3]
```diff
@@ -21,20 +21,24 @@
         android:layout_width="match_parent"
         android:layout_height="?attr/actionBarSize"
         android:background="?attr/colorPrimary"
```
Comment (rt4914) [inline app/src/main/res/layout/topic_fragment.xml]: Also change `colorPrimary` in `colors.xml` file, as this color is currently has issue related to contrast in accessibility scanner. Use: `#00645C` color mentioned in V7 Mocks.

**Review by rt4914 — APPROVED** [review]
Once changes are done, run accessibility scanner to check everything.

Also, one more point: The title of the toolbar is `Topic: <TOPIC_NAME>`

**Comment by nikitamarysolomanpvt** [comment]
@mschanteltc Please find the below issues.
screenshot_Oppia_2019-11-13-20_01_17

1. Text contrast
[358,336][451,389]
The item's text contrast ratio is 4.34. This ratio is based on an estimated foreground color of #B3D1CE and an estimated background color of #00645C. Consider using a contrast ratio greater than 4.50 for small text, or 3.00 for large text.

2.  Text contrast
[618,336][731,389]
The item's text contrast ratio is 4.34. This ratio is based on an estimated foreground color of #B3D1CE and an estimated background color of #00645C. Consider using a contrast ratio greater than 4.50 for small text, or 3.00 for large text.

4. I feel the icons given for tabs are not actually matching the mock. Please provide svg icons 24/ 24 height/width and 24 * 24 viewport

File: app/src/main/res/values/styles.xml (hunk 2 above)
Comment (nikitamarysolomanpvt) [reply app/src/main/res/values/styles.xml]: Done

File: app/src/main/res/layout/topic_fragment.xml (hunk 3 above)
Comment (nikitamarysolomanpvt) [reply app/src/main/res/layout/topic_fragment.xml]: You want me to change colorPrimary to #00645C??

Comment (rt4914) [reply app/src/main/res/layout/topic_fragment.xml]: Yes.

File: app/src/main/res/values/styles.xml (hunk 1 above)
Comment (nikitamarysolomanpvt) [reply app/src/main/res/values/styles.xml]: Done

File: app/src/main/res/layout/topic_fragment.xml (hunk 3 above)
Comment (nikitamarysolomanpvt) [reply app/src/main/res/layout/topic_fragment.xml]: used colorPrimaryDark

**Comment by rt4914** [comment]
@nikitamarysolomanpvt why have I been assigned again? I am not able to see any changes in this.

**Comment by nikitamarysolomanpvt** [comment]
> @nikitamarysolomanpvt why have I been assigned again? I am not able to see any changes in this.

Sorry i merged the code with an old branch which .PTAL now

**Comment by rt4914** [comment]
> Once changes are done, run accessibility scanner to check everything.
> 
> Also, one more point: The title of the toolbar is `Topic: <TOPIC_NAME>`

@nikitamarysolomanpvt please make this change.

**Review by rt4914 — APPROVED** [review]
There is one change remaining, check all the comments.

File: app/src/main/res/layout/topic_fragment.xml [hunk 4]
```diff
@@ -20,21 +20,25 @@
         android:id="@+id/toolbar"
         android:layout_width="match_parent"
         android:layout_height="?attr/actionBarSize"
-        android:background="?attr/colorPrimary"
-        app:titleTextColor="@color/white"
-        app:popupTheme="@style/ThemeOverlay.AppCompat.Light" />
+        android:background="@color/oppiaPrimaryDark"
+        android:textSize="20dp"
```
Comment (veena14cs) [inline app/src/main/res/layout/topic_fragment.xml:24]: make it 20sp

**Review by veena14cs — APPROVED** [review]
Looks good please address one comment.

**Comment by mschanteltc** [comment]
> @mschanteltc Please find the below issues.
> screenshot_Oppia_2019-11-13-20_01_17
> 
> 1. Text contrast
>    [358,336][451,389]
>    The item's text contrast ratio is 4.34. This ratio is based on an estimated foreground color of #B3D1CE and an estimated background color of #00645C. Consider using a contrast ratio greater than 4.50 for small text, or 3.00 for large text.
> 2. Text contrast
>    [618,336][731,389]
>    The item's text contrast ratio is 4.34. This ratio is based on an estimated foreground color of #B3D1CE and an estimated background color of #00645C. Consider using a contrast ratio greater than 4.50 for small text, or 3.00 for large text.
> 3. I feel the icons given for tabs are not actually matching the mock. Please provide svg icons 24/ 24 height/width and 24 * 24 viewport

For the header text, the font is pure white #FFFFFF while the background is #00645C. This would provide a contrast ratio of [7.05](https://webaim.org/resources/contrastchecker/?fcolor=FFFFFF&bcolor=00645C).

For the unselected tab objects, they are all #FFFFFF but have different opacities. The selected that will be 100% opacity. The unselected tabs will increase opacity from 70% to 80%. Using an eyedropper tool, this will make the icons appear as #CCE0DE (do **not** use this hex for sake of consistency). The contrast this will have against the background would be [5.13](https://webaim.org/resources/contrastchecker/?fcolor=CCE0DE&bcolor=00645C).

I will post the icons in the shared Google Drive folder and will let you know when they are available.

**Comment by mschanteltc** [comment]
I shared the icons in the Google Drive folder called "Topic Page."

**Comment by nikitamarysolomanpvt** [comment]
> I shared the icons in the Google Drive folder called "Topic Page."

Done. But still i feel two of them looks big other two looks smaller comparatively.

File: app/src/main/res/layout/topic_fragment.xml (hunk 4 above)
Comment (nikitamarysolomanpvt) [reply app/src/main/res/layout/topic_fragment.xml:24]: Done

**Comment by mschanteltc** [comment]
Is it possible you can share a screenshot so I can understand which two needs fixing please?

**Pull request merged**